### Silhouette Index

In [2]:
import pandas as pd
dataset = pd.read_csv("Mall_Customers.csv")
X = dataset.iloc[:, [3,4]].values
from sklearn.cluster import KMeans
from sklearn.metrics import silhouette_score
noofclus = 16


for i in range(2, noofclus):
    k_means = KMeans(n_clusters = i, init = 'k-means++', random_state = 42)
    k_means.fit(X)
    labels = k_means.predict(X)
    print(i, silhouette_score(X, labels))



2 0.2968969162503008
3 0.46761358158775435
4 0.4931963109249047
5 0.553931997444648
6 0.5379675585622219
7 0.5288104473798049
8 0.45924987052000815
9 0.4565077334305076
10 0.45056557470336733
11 0.437841976811071
12 0.4237175596671194
13 0.4211574151725363
14 0.41599158155673244
15 0.43212006235899736


Silhouette index close to 1 is good. It means in the above example, the index value 0.5539 for no of clusters 5 is appropriate.